### PR TITLE
Generate annotations for extension fields

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

**Comment by swankjesse** [comment]
Code changes only in the first commit:
https://github.com/square/wire/pull/1705/commits/8253c22508c741638c25462896d24e1745eb5bd7

**Comment by swankjesse** [comment]
My goal is for application developers to write this:

```
extend google.protobuf.FieldOptions {
  optional string documentation_url = 60001;
}
```

```
message Receipt {
  repeated string loyalty = 1 [documentation_url ="https://squareup.com/us/en/software/loyalty"];
}
```

to get this:

```
  @WireField(
      tag = 1,
      adapter = "com.squareup.wire.ProtoAdapter#SFIXED32"
  )
  @DocumentationUrl("https://squareup.com/us/en/software/loyalty")
  public final List<String> loyalty;
```

**Comment by JakeWharton** [comment]
I'm going to look tomorrow, but is there a test for when the option is a complex message type where you'd want both forms?

**Comment by swankjesse** [comment]
I totally punted on that. Only emitting annotations if the field is a scalar or an enum. Keeps it simple, and addresses most use-cases.

File: wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/SyntaxRules.kt [hunk 1]
```diff
@@ -106,7 +106,7 @@ interface SyntaxRules {
         return Field.EncodeMode.OMIT_IDENTITY
       }
 
-      override fun jsonName(name: String): String = camelCase(name)
+      override fun jsonName(name: String): String = camelCase(name, false)
```
Comment (oldergod) [inline wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/SyntaxRules.kt:109]:
```suggestion
      override fun jsonName(name: String): String = camelCase(name, upperCamel  = false)
```

File: wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/SchemaTest.kt [hunk 2]
```diff
@@ -560,7 +560,7 @@ class SchemaTest {
       assertThat(expected).hasMessage("""
             |unable to resolve foo_package.Foo
             |  for field unknown (/source/message.proto:4:3)
-            |  in message Message (/source/message.proto:1:1)
+            |  in extend Message (/source/message.proto:3:1)
```
Comment (oldergod) [inline wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/SchemaTest.kt:563]: Nice

File: wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/SyntaxRules.kt (hunk 1 above)
Comment (swankjesse) [reply wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/SyntaxRules.kt:109]: Fix in follow-up!

**Pull request merged**